### PR TITLE
[AUD-1630] Add UploadTrack button and drawer

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Adds UploadTrack button and refactors MobileUploadDrawer
- Adds commonAlt button variant
- Adds GradientIcon component
- Adds fontSize to typography

https://user-images.githubusercontent.com/8230000/156069871-c0cbdc2c-e40f-438c-b064-e7a8b35aed96.mov

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-aud-1630-add-upload-track-drawer

File: packages/mobile/src/components/core/Button.tsx [hunk 1]
```diff
@@ -57,12 +57,26 @@ const useStyles = makeStyles(
...
+          color: palette.neutral
+        }
       }
     }
 
     const variantPressingStyles = {
       secondary: variantStyles.primary,
-      common: variantStyles.primary
+      common: variantStyles.primary,
+      commonAlt: variantStyles.commonAlt
```
Comment (raymondjacobson) [inline packages/mobile/src/components/core/Button.tsx:79]: yes pls

File: packages/mobile/src/components/core/GradientIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { ComponentType } from 'react'
```
Comment (raymondjacobson) [inline packages/mobile/src/components/core/GradientIcon.tsx:1]: Note: https://github.com/AudiusProject/audius-client/blob/main/packages/mobile/src/components/enable-push-notifications-drawer/EnablePushNotificationsDrawer.tsx#L145

I believe this does what you're doing here.
I like GradientIcon though.

**Review by raymondjacobson — COMMENTED** [review]
Realize now, we already have this drawer

https://github.com/AudiusProject/audius-client/blob/main/packages/mobile/src/components/mobile-upload-drawer/MobileUploadDrawer.tsx#L29

File: packages/mobile/src/components/core/GradientIcon.tsx (hunk 2 above)
Comment (dylanjeffers) [reply packages/mobile/src/components/core/GradientIcon.tsx:1]: > Realize now, we already have this drawer
> 
> https://github.com/AudiusProject/audius-client/blob/main/packages/mobile/src/components/mobile-upload-drawer/MobileUploadDrawer.tsx#L29

Lol im so dumb! took not that long at least :D will merge them

**Review by raymondjacobson — APPROVED** [review]

**Pull request merged**